control_service/control_service/routers/users.py

```diff
@@ -0,0 +1,141 @@
+
+async def get_user(username: str):
+    query = sql_userdata.select().where(sql_userdata.c.UserName==username)
+    result= await database.fetch_one(query)
+    print(result)
+    if result:
+        return result
+
+async def authenticate_user( username: str, password: str):
+    user = await get_user(username)
+    if not user:
+        return False
+    if not verify_password(password, user.password):
+        return False
+    return user
+
+
+def create_access_token(*, data: dict, expires_delta: timedelta = None):
+    to_encode = data.copy()
+    if expires_delta:
+        expire = datetime.utcnow() + expires_delta
+    else:
+        expire = datetime.utcnow() + timedelta(minutes=15)
+    to_encode.update({"exp": expire})
+    jwt.encode(to_encode, SECRET_KEY)
+    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+    return encoded_jwt
+
+async def get_current_user(token: str = Depends(oauth2_scheme)):
+    credentials_exception = HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Could not validate credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+    try:
+        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
+        username: str = payload.get("sub")
+        if username is None:
+            raise credentials_exception
+        token_data = TokenData(username=username)
+    except jwt.PyJWTError:
+        raise credentials_exception
+    user = await get_user(username=token_data.username)
+    if user is None:
+        raise credentials_exception
+    return user
+
+async def get_current_active_user(current_user: UserData = Depends(get_current_user)):
+    if current_user.Enabled is False:
+        raise HTTPException(status_code=403, detail="Inactive user")
+    return current_user
+
+
+@router.post("/Login", response_model=Token)
+async def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends()):
+    user = await authenticate_user(form_data.username, form_data.password)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+    if user.Enabled is False:
+        raise HTTPException(status_code=403, detail="Inactive user")
+    access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+    access_token = create_access_token(
+        data={"sub": user.UserName}, expires_delta=access_token_expires
+    )
+    return {"access_token": access_token, "token_type": "bearer"}
+
+
+@router.post('/Register')
+async def register(data: RegisterUser):
+
+    response = {}
+    query = sql_userdata.select().where(sql_userdata.c.UserName == data.UserName)
+    query2 = sql_userdata.select().where(sql_userdata.c.Email == data.Email)
+    result = await database.fetch_one(query)
+    result2 = await database.fetch_one(query2)
```

Comment:
You could use [create_task](https://docs.python.org/3/library/asyncio-task.html#asyncio.create_task) to wait for the database opperations concurrently.